drivers/edac/shell.c

```diff
@@ -71,27 +72,38 @@ static int cmd_edac_info(const struct shell *shell, size_t argc, char **argv)
 
 	shell_fprintf(shell, SHELL_NORMAL, "Show EDAC status\n");
 
-	error = edac_ecc_error_log_get(dev);
+	ret = edac_ecc_error_log_get(dev, &error);
+	if (ret) {
```

Comment:
Assuming this is under the scope that the coding style will be enforced on, I think only boolean expressions or bool type variables can be used as if-conditions, so this would have to be `if (ret != 0)`. @nashif can confirm.

Reply:
Ok, I see @nashif reaction, will change to boolean.